ui/src/components/ToS.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+import {LitElement, html, css} from 'lit-element';
+
+import {ToSService} from '../utils/ToSService';
+
+import style from '../styles/components/ToS.css.js';
+
+import {ListItem} from '@material/mwc-list/mwc-list-item';
+ 
+/**
+ * Component responsible for providing users a means to agree
+ * to the terms of service. 
+ */
+export class ToS extends LitElement {
+
+    static get properties() {
+        return {
+            country: {type: String},
+            tos: {type: Array},
+            disabled: {type: Boolean}
+        };
+    }
+
+    static get styles() {
+        return style;
+    }
+
+    constructor() {
+        super();
+        this.tos = [];
+        this.disabled = false;
+        this.tosService = new ToSService(); 
+    }
+
+    async firstUpdated() {
+        this.tos = await this.tosService.getToS(this.country);
+    }
+
+    updated() {
+        this.detectOverflow(); 
+    }
+
+    /**
+     * Enables the 'accept-button' if it has been detected that the user has read the terms of service
+     * in its entirety. 
+     */
+    checkScroll() {
```

Comment:
It may be that you have to turn checkScroll into a bound function

private readonly checkScroll = () => {};

When you pass in checkScroll to the mwc-button it changes the scope of the "this" in "this.disabled" to the button element.